python_modules/dagster/dagster/_core/definitions/multi_dimensional_partitions.py

```diff
@@ -325,7 +327,13 @@ def __init__(
     ):
         check.inst_param(partitions_def, "partitions_def", MultiPartitionsDefinition)
         subset = (
-            set(partitions_def.get_partition_key_from_str(key) for key in subset)
```

Comment:
If a user has existing static partition backfills, changing the asset's partitions def to a multipartitioned asset would raise an error. This is because a multipartitions subset is serialized the same way as a default partitions subset, so we can't distinguish between the two using the serialized content. This line would then error because the static partition keys don't contain the "|" delimiter.

This change would enable existing static partition backfills to load even after the partitions def has been changed to a multipartitions def, by filtering out any invalid keys here

Reply:
FWIW, I think it's fine if you can't load a backfill after moving from static to multi-partitioned.

Reply:
The struggle is that the whole backfill page won't load if a single backfill can't be deserialized, which is the reason I'm trying to support this situation. 

Maybe @bengotow knows how to change the backfill page to avoid breaking the whole page if a single row can't be deserialized?